mlserver/parallel/pool.py

```diff
@@ -117,7 +119,10 @@ async def predict(
             model_version=model_version,
             inference_request=inference_request,
         )
-        self._requests.put(request_message)
```

Comment:
I am interested to know why we think round robin in more effective that having a single queue especially if one of the workers is very busy?

I am assuming that if one worker is very busy, the other workers will fetch work from the shared queue, which will make the system have an overall better throughput?

Reply:
Ah I see, a worker can basically consume all work from the queue and convert them to coroutines that are waiting to execute on the worker event loop? can we have a  limit, perhaps if the current count of gorouttines on a specific worker is beyond a certain threshold then the worker is not able to pull more from the queue? Having one queue per-worker enables more smart scheduling decisions so I think this is the correct approach and given what you described, round robin is going to spread load across workers evenly which we didnt have before. Maybe a future PR would schedule work to the queue with least messages, assuming we can control how the worker from consuming from the queue?

Reply:
Yuuup, totally. 

In the future, it would make sense to choose the worker based on its backlog. Since each worker has its own queue, this shouldn't be too hard TBH.